task/init/0.2/migrations/0.2.1.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Created for task: init@0.2.1
+# Creation time: 2025-09-19T10:00:53Z
+
+declare -r pipeline_file=${1:?missing pipeline file}
+
+
+buildah_format="docker"
+# Check if the run-opm-command task exists
+if yq -e '.spec.tasks[] | select(.name == "run-opm-command")' "$pipeline_file" >/dev/null; then
+    # FBC builder needs to be OCI by default
+    echo "FBC pipeline detected, setting format to oci"
+    buildah_format="oci"
+fi
+
+if yq -e '.spec.tasks[] | select(.taskRef.params[] | (.name == "name" and (.value == "tkn-bundle" or .value == "tkn-bundle-oci-ta")))' "$pipeline_file" >/dev/null; then
+    # Tekton bundle should be OCI
+    echo "Tekton bundle pipeline detected, setting format to oci"
+    buildah_format="oci"
+fi
+
+
+# Determine all tasks which should be updated
+tasks_to_be_updated=()
+
+if yq -e '.spec.tasks[] | select(.name == "build-images")' "$pipeline_file" >/dev/null; then
+    tasks_to_be_updated+=( "build-images" )
+fi
+
+if yq -e '.spec.tasks[] | select(.name == "build-container")' "$pipeline_file" >/dev/null; then
+    tasks_to_be_updated+=( "build-container" )
+fi
```

Comment:
I think we should include these task names only if they use the buildah task ref. Buildah and build-image-index are the only ones that have the BUILDAH_FORMAT param, so it wouldn't make sense to add the param if it's a different task.

That way we would exclude the tekton-bundle-builder pipeline, for example (which we do want to exclude IMO)

Reply:
reworked